### PR TITLE
Add controller scaffolding

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/14270

### CODE REVIEW

File: controllers/deployment_validator.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	if err != nil {
+		v.Log.With("stack", microerror.JSON(err)).Error("failed with error")
+		return admission.Errored(500, err)
+	}
+
+	return resp
+}
+
+func (v *DeploymentValidator) handle(ctx context.Context, req admission.Request, deployment *appsv1.Deployment) (admission.Response, error) {
+	return admission.Denied("__NOT_IMPLEMENTED__"), nil
```
Comment (kopiczko) [inline controllers/deployment_validator.go:56]: The actual implementation will go here.

File: config/webhook/manifests.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
```
Comment (kopiczko) [inline config/webhook/manifests.yaml:1]: This file is generated from `//+kubebuilder:webhook` marker. The rest of files in `config/webhook` were copied from a kubebuilder-style webhook I generated on a side.

File: main.go [hunk 3]
```diff
@@ -98,7 +98,14 @@ func main() {
...
-	//controllersLog := rootLog.Named("controllers")
+	controllersLog := rootLog.Named("controllers")
+	if err = (&controllers.DeploymentValidator{
+		Client: mgr.GetClient(),
+		Log:    controllersLog.Named("deployment-validator").Sugar(),
+	}).SetupWebhookWithManager(mgr); err != nil {
+		setupLog.Error("unable to create webhook", zap.NamedError("stack", err))
+		os.Exit(1)
+	}
 	//+kubebuilder:scaffold:builder
```
Comment (kopiczko) [inline main.go:109]: I wrote this wiring manually but it's ~compatible with kubebuilder style wiring.

File: config/rbac/role.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (kopiczko) [inline config/rbac/role.yaml:1]: This is generated from `// +kubebuilder:rbac` marker.

File: config/default/webhookcainjection_patch.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# This patch add annotation to admission webhook config and
```
Comment (kopiczko) [inline config/default/webhookcainjection_patch.yaml:1]: Copied from a kubebuilder-style webhook I generated on a side

File: config/default/manager_webhook_patch.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+apiVersion: apps/v1
```
Comment (kopiczko) [inline config/default/manager_webhook_patch.yaml:1]: Copied from a kubebuilder-style webhook I generated on a side

**Review by corest — APPROVED** [review]

**Pull request merged**